opentelemetry/src/main/scala/zio/telemetry/opentelemetry/Tracing.scala

```diff
@@ -98,6 +126,24 @@ object Tracing {
       r   <- createChildOf(old, spanName, spanKind).use(finalizeSpanUsingEffect(effect, _, toErrorStatus))
     } yield r
 
+  /**
+   * Sets the current span to be the child of the current span with name
+   * 'spanName', introducing a thread-local scope during the execution allowing
+   * non-zio context propagation. Ends the span (and scope) when the effect
+   * finishes.
+   */
+  def scopedSpan[R, E, A](
+    spanName: String,
+    spanKind: Span.Kind,
+    toErrorStatus: PartialFunction[E, Status]
+  )(effect: ZIO[R, E, A]): ZIO[R with Tracing with Blocking, E, A] =
```

Comment:
Do we really need the blocking ExecutionContext here, or would an uninterruptible effect be sufficient?

Reply:
Chances are you would know better than I! 

The way I was reading the docs, in order to guarantee staying on the same thread requires a .lock call, which requires an EC, and the blocking context was the easiest way to get access to one. I'll try dropping blocking and using `.uninterruptable` instead and see if the tests pass -- they do not pass with the `blocking` removed entirely.

Reply:
I've switched to uninterruptible, which is still passing the tests. I will do some more testing outside and see how it works. Interestingly, most recently the tests pass without anything, but I don't think they create a sufficiently complex environment to cause interruptions consistently.

Reply:
I've switched the implementation to match how the upstream project handles this, by wrapping the runnables. This eliminates the need for blocking or uninterruptible effects in favor of locking the executor with a wrapped copy of the same executor that wraps all the runnables it sees.